frontend/src/App.vue

```diff
@@ -54,16 +53,67 @@ export default {
       );
     },
   },
-  watch: {
-  },
+  watch: {},
   methods: {
     async onAction(contextualItem, index) {
       const itemFqid = contextualItem?.item?.fqid;
-      await this.rpc.invoke("performAction", [itemFqid, index, contextualItem.context]);
+      await this.rpc.invoke("performAction", [
+        itemFqid,
+        index,
+        contextualItem.context,
+      ]);
     },
+  getItem(itemFqid) {
+    for (const collection of this.collections) {
+      for (const item of collection.items) {
+        if (item.fqid === itemFqid) {
+          return item;
+        }
+        if (item.items) {
+          for (const subItem of item.items) {
+            if (subItem.fqid === itemFqid) {
+              return subItem;
+            }  
+            else if( subItem.items ){
+              const deepSubitem =  this.getDeepSubitem(itemFqid, subItem);
+              if(deepSubitem){
+                return deepSubitem;
+              }
+            }
+          }
+        }
+      }
+    }
+    // TODO - console log: item does not exist
+  },
+ getDeepSubitem( itemFqId, item ){
+    if( item.items ){
+      for (const subItem of item.items) {
+        if (subItem.fqid === itemFqId) {
+          return subItem;
+        } else if(subItem.items){
+            return this.getDeepSubitem(itemFqId, subItem);
```

Comment:
It seems to me that:
this `return` will **break** the outer loop which means
that if the Nth subItem has nested items, any following items in `item.items` (line 91) won't be searched at all